src/Libraries/Microsoft.Extensions.AI/TextToImage/TextToImageClientBuilderTextToImageClientExtensions.cs

```diff
@@ -0,0 +1,38 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.Shared.Diagnostics;
+
+namespace Microsoft.Extensions.AI;
+
+/// <summary>Provides extension methods for working with <see cref="ITextToImageClient"/> in the context of <see cref="TextToImageClientBuilder"/>.</summary>
+[Experimental("MEAI001")]
+public static class TextToImageClientBuilderTextToImageClientExtensions
+{
+    /// <summary>Creates a new <see cref="TextToImageClientBuilder"/> using <paramref name="innerClient"/> as its inner client.</summary>
+    /// <param name="innerClient">The client to use as the inner client.</param>
+    /// <returns>The new <see cref="TextToImageClientBuilder"/> instance.</returns>
+    /// <remarks>
+    /// This method is equivalent to using the <see cref="TextToImageClientBuilder"/> constructor directly,
+    /// specifying <paramref name="innerClient"/> as the inner client.
+    /// </remarks>
+    public static TextToImageClientBuilder AsBuilder(this ITextToImageClient innerClient)
+    {
+        _ = Throw.IfNull(innerClient);
+
+        return new TextToImageClientBuilder(innerClient);
+    }
+
+    /// <summary>Adds the provided <see cref="ITextToImageClient"/> to the text to image client builder as the outermost client in the pipeline.</summary>
+    /// <param name="builder">The <see cref="TextToImageClientBuilder"/>.</param>
+    /// <param name="client">The <see cref="ITextToImageClient"/> to add to the pipeline.</param>
+    /// <returns>The <paramref name="builder"/>.</returns>
+    public static TextToImageClientBuilder Use(this TextToImageClientBuilder builder, ITextToImageClient client)
```

Comment:
This design looks stale, e.g. see https://github.com/dotnet/extensions/blob/5d8efb684ac5a56965f6fb72037cbd1db51ad74e/src/Libraries/Microsoft.Extensions.AI/Embeddings/EmbeddingGeneratorBuilder.cs#L47

Reply:
Can you elaborate on what parts I should add?

I did a diff with embeddings (instead of speech-to-text) and found I was missing AnonymousDelegatingTextToImageClient, CachingTextToImageClient, DistributedCachingTextToImageClient, and OpenTelemetryTextToImageClient.

I'm not sure "Caching" ones make sense here.  OTel is something we might want to add - I noticed that was missing when trying to buidl the sample [ericstj/textToImageSample](https://github.com/ericstj/textToImageSample)